File: tests/fixtures/_highlight-corpus/sample.cs
// C# sample
using System;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Demo;

public sealed class Counter
{
    private int _value;

    public int Value => _value;

    public int Increment(int n = 1)
    {
        _value += n;
        return _value;
    }
}

internal static class Program
{
    public static async Task<int> Main()
    {
        var lines = await File.ReadAllLinesAsync("input.txt");
        var counter = new Counter();
        foreach (var line in lines.Where(l => !string.IsNullOrEmpty(l)))
        {
            counter.Increment(line.Length);
        }
        Console.WriteLine($"total bytes: {counter.Value}");
        return 0;
    }
}

File: tests/fixtures/_highlight-corpus/sample.rb
# Ruby sample
# frozen_string_literal: true

class Counter
  attr_reader :value

  def initialize
    @value = 0
  end

  def increment(n = 1)
    @value += n
    @value
  end
end

def main
  counter = Counter.new
  File.foreach('input.txt') do |line|
    line = line.chomp
    counter.increment(line.length) unless line.empty?
  end
  puts "total bytes: #{counter.value}"
end

main if __FILE__ == $PROGRAM_NAME
